src/pymatgen/analysis/graphs.py

```diff
@@ -933,7 +933,7 @@ def draw_graph_to_file(
                     d["label"] = f"{d['weight']:.2f} {units}"
 
             # update edge with our new style attributes
-            g.edges[u, v, k] |= d
+            g.edges[u, v, k].update(d)
```

Comment:
@janosh Forgot to mention this, I have to revert your `.update -> |=` change here c67abbc6678ebeee6a324440570aa38cc0c66adb because it seems to [break unit test](https://github.com/materialsproject/pymatgen/actions/runs/10246364887/job/28343300325). 

I haven't got a chance to look closer yet, perhaps the operator is not defined for `edges`

Reply:
yeah i noticed that, that was sloppy refactoring on my part. thanks for fixing!

> perhaps the operator is not well defined for edges

that's very likely correct

Reply:
> that was sloppy refactoring on my part.

`pygraphviz` was not installed in CI at that time, so there's no way for you to catch that :)

Reply:
no, but i could have noticed that `g.edges[u, v, k]` is unlikely to be a `dict` in which the first thought is to check whether the class in question implements `__or__`. (if not, `|=` is undefined behavior)